### PR TITLE
Support bech 32

### PR DESCRIPTION
**Pull request by KotlinGeekDev**
Sorry for all the mess. :sweat_smile:

### CODE REVIEW

**Comment by KotlinGeekDev** [comment]
cc @Giszmo

File: nostrpostrlib/src/test/java/nostr/postr/UtilsTest.kt [hunk 1]
```diff
@@ -37,4 +37,19 @@ internal class UtilsTest {
...
+        val pubkeyHex = "e4c47aedea8ea54255f5ba07a77053b24553e9b975435e56da343da19aec7881"
+        val npubRepresentation = Hex.decode(pubkeyHex).toNpub()
+        val correctRep = "npub1unz84m0236j5y404hgr6wuznkfz486dew4p4u4k6xs76rxhv0zqsq9q6t8"
+
+        val jackPubkeyHex = "82341f882b6eabcd2ba7f1ef90aad961cf074af15b9ef44a09f9d2a8fbfbe6a2"
+        val derivedJackNpub = Hex.decode(jackPubkeyHex).toNpub()
+        val correctJackNpub = "npub1sg6plzptd64u62a878hep2kev88swjh3tw00gjsfl8f237lmu63q0uf63m"
+
+        assertEquals(correctRep, npubRepresentation)
+        assertEquals(correctJackNpub, derivedJackNpub)
```
Comment (Giszmo) [inline nostrpostrlib/src/test/java/nostr/postr/UtilsTest.kt]: Would you mind sharing where the test vectors are from? And privkey would also be nice to have.

File: nostrpostrlib/src/main/java/nostr/postr/Bech32Util.kt [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+//        val version = data[0]
+//        require(version in 0..16) { "invalid segwit version" }
+//        val bin = five2eight(data, 1)
+//        require(bin.size in 2..40) { "invalid witness program length ${bin.size}" }
+//        if (version == 0.toByte()) require(encoding == Encoding.Bech32) { "version 0 must be encoded with Bech32" }
+//        if (version == 0.toByte()) require(bin.size == 20 || bin.size == 32) { "invalid witness program length ${bin.size}" }
+//        if (version != 0.toByte()) require(encoding == Encoding.Bech32m) { "version 1 to 16 must be encoded with Bech32m" }
+//        return Triple(hrp, version, bin)
+//    }
+
```
Comment (Giszmo) [inline nostrpostrlib/src/main/java/nostr/postr/Bech32Util.kt]: No dead code, please. If there is any chance we might need this soon, add it. If not (I think we don't), remove it.

```suggestion
```

**Review by Giszmo — CHANGES_REQUESTED** [review]
Thanks for PR. Two tiny changes please.

File: nostrpostrlib/src/test/java/nostr/postr/UtilsTest.kt (hunk 1 above)
Comment (KotlinGeekDev) [reply nostrpostrlib/src/test/java/nostr/postr/UtilsTest.kt]: OK No problem. The first one, I got from a person who converted their hex-formatted key to bech32. And the second is Jack Dorsey's pubkey.

File: nostrpostrlib/src/main/java/nostr/postr/Bech32Util.kt (hunk 2 above)
Comment (KotlinGeekDev) [reply nostrpostrlib/src/main/java/nostr/postr/Bech32Util.kt]: No problem.

**Comment by KotlinGeekDev** [comment]
Adressed issues. On the subject of privkeys, I could not find any test data. Or should I look at the test data you used for the privkey(), pubkey() tests?

File: nostrpostrlib/src/test/java/nostr/postr/UtilsTest.kt (hunk 1 above)
Comment (Giszmo) [reply nostrpostrlib/src/test/java/nostr/postr/UtilsTest.kt]: Can you add that as a code comment? Link to the tool used to convert for example.

**Review by Giszmo — APPROVED** [review]

**Pull request merged**